layers/+tools/shell/packages.el

```diff
@@ -219,6 +219,10 @@
              (t (comint-simple-send proc command))))))
   (add-hook 'shell-mode-hook 'shell-comint-input-sender-hook)
   (add-hook 'shell-mode-hook 'spacemacs/disable-hl-line-mode)
+
+  ;; bind M-r to helm if possible
+  (spacemacs//shell-bind-m-r shell-mode-map)
```

Comment:
Whats the current default behaviour? If it is highly different to the helm/ivy based approach you should have a config var to activate it. The default would be nil by default meaning no new behaviour, t to activate completion support.

This is especially true as the default history file may be wrong for non bash shells like fish or zsh.